Morphic.Settings.Tests/RegistrySettingHandlerTests.cs

```diff
@@ -0,0 +1,175 @@
+        [InlineData(Setting.ValueKind.Boolean, RegistryValueKind.DWord, typeof(Int64), "whoopsie", true, false)]
+        public async Task TestCapture(Setting.ValueKind kind, RegistryValueKind rkind, System.Type type, object registryval, object finalval, bool success)
+        {
+            var registry = new MockRegistry();
+            var loggerFactory = new LoggerFactory();
+            var logger = loggerFactory.CreateLogger<RegistrySettingHandler>();
+
+            var setting = new Setting()
+            {
+                Name = "test",
+                Kind = kind,
+                HandlerDescription = new RegistrySettingHandlerDescription(@"HKEY_TEST\Test\Key", "SomeValue", rkind)
+            };
+            var handler = new RegistrySettingHandler(setting, registry, logger);
+
+            var result = await handler.Capture();
+            Assert.False(result.Success);   //will fail due to null
+            Assert.Null(result.Value);
+
+
+            var callCount = 0;
+            registry.NextGetResponder = (string keyName, string valueName, object? defaultValue) =>
+            {
+                ++callCount;
+                Assert.Equal(@"HKEY_TEST\Test\Key", keyName);
+                Assert.Equal(@"SomeValue", valueName);
+                Assert.Null(defaultValue);
+                return registryval;
+            };
+            result = await handler.Capture();
+            Assert.Equal(1, callCount);
+            if (success)
+            {
+                Assert.True(result.Success);
+                Assert.NotNull(result.Value);
+                Assert.Equal(result.Value!.GetType(), type);
+                Assert.Equal(finalval, result.Value);
+            }
+            else
+            {
+                Assert.False(result.Success);
+                Assert.Null(result.Value);
+            }
+        }
+
+        [Theory]
+        [InlineData(Setting.ValueKind.String, RegistryValueKind.String, typeof(string), "Hello", "Hello", true)]
+        [InlineData(Setting.ValueKind.Integer, RegistryValueKind.DWord, typeof(Int64), 52L, 52L, true)]
+        [InlineData(Setting.ValueKind.Boolean, RegistryValueKind.DWord, typeof(Int64), "whoopsie", true, false)]
+        public async Task TestApply(Setting.ValueKind kind, RegistryValueKind rkind, System.Type type, object initval, object registryval, bool success)
+        {
+            var registry = new MockRegistry();
+            var loggerFactory = new LoggerFactory();
+            var logger = loggerFactory.CreateLogger<RegistrySettingHandler>();
+
+            var setting = new Setting()
+            {
+                Name = "test",
+                Kind = kind,
+                HandlerDescription = new RegistrySettingHandlerDescription(@"HKEY_TEST\Test\Key", "SomeValue", rkind)
+            };
+            var handler = new RegistrySettingHandler(setting, registry, logger);
+
+            var pass = await handler.Apply("World");    //should fail, no callback
```

Comment:
Similar to above, make it explicit what is being tested here by setting `registry.NextSetResponder`.  Also need to test all exceptions that `Win32.Registry` might throw.  Consider splitting into separate test case